bolt-simple-taproot.md

```diff
@@ -0,0 +1,958 @@
+
+Note that nonces can safely be pre-generated, and even batched/pipelined ahead
+of time to reduce the number of round trips in the protocol.
+
+Finally, due to the asymmetric state of the current revocation channels, we
+actually require two sets of public nonces to be exchanged by each party: one
+for the local commitment, and one for the remote commitment.
+
+#### Signing
+
+Once public nonces have been exchanged, the `NonceAgg` algorithm is used to
+combine the public nonces into the combined nonce which will be a part of the
+final combined signature.
+
+The `Sign` is used to generate a valid partial signature, with the
+`PartialSigVerify` algorithm used to verify each signature.
+
+Once all partial signatures are known, the `PartialSigAgg` algorithm is used to
+combine the partial signature into a final valid BIP 340 schnorr signature.
+
+## Design Overview
+
+With the preliminaries out of the way, we provide a brief overview of the
+Simple Taproot Channel design.
+
+The multi-sig output becomes a single P2TR key, with `musig2` key aggregation
+used to combine two keys into one. During the funding flow, public nonces are
+exchanged with the first two messages.
+
+For commitment transactions, we inherit the anchor outputs semantics, meaning
+there're two outputs used for CPFP, with all scripts inheriting a `CSV 1`
+restriction.
+
+The local output of the commitment transaction uses the revocation key as the
+taproot internal key, and commits to a single script path of the delay script.
+
+The remote output of the commitment transaction uses the `simple_taproot_nums`
+as the top-level internal key, and then commits to a normal remote script.
+
+Anchor outputs use the respective local and remote keys from the commitment
+transaction as the top level output, with the committed script being the `CSV
+16` delay.
+
+All HTLC scripts use the revocation key as the top-level internal key. This
+allows the revocation to be executed without additional on-chain bytes, and
+also reduces the amount of state nodes need to keep in order to properly handle
+channel breaches.
+
+## Specification
+
+### Feature Bits
+
+Inheriting the structure put forth in BOLT 9, we define a new feature bit to be
+placed in the `init` message, `node_announcement` and also
+`channel_announcement`. We opt to also place the bit in the
+`channel_announcement` so taproot channels can be identified on the routing
+level, which will eventually matter once PTLCs are fully rolled out.
+
+
+| Bits  | Name                             | Description                                               | Context  | Dependencies      | Link                                  |
+|-------|----------------------------------|-----------------------------------------------------------|----------|-------------------|---------------------------------------|
+| 50/51 | `option_simple_taproot`| Node supports simple taproot chanenls | IC | `option_channel_type`+`option_anchors` | TODO(roasbeef): link |
```

Comment:
does 50/51 not conflict with option_zeroconf?

Reply:
Now 80/81 and also 180/181 (staging bit).